Add Tox and Github Actions CI support

luzfcb opened the pull request:
- Add `tox.ini` configuration to be able to test locally on all supported versions.
- Add support to run tests on Github Actions (Github Actions offers free 2,000 Actions minutes/month per free GitHub account. In paid accounts, this amount may be higher)
- Add `python3.9` to the test matrix. Fix #180
- Add Django 3.1 and Django 3.2rc1 to the test matrix. Fix #182
- Add `djangorestframework` 3.11 and 3.12 to the test matrix. Fix #184

A sample of how the Github Action will look like:
Also, the sample of full  Github Action test log of this PR can be accessed in https://github.com/luzfcb/django-versatileimagefield/actions/runs/692862505


![image](https://user-images.githubusercontent.com/807599/112723057-7ec17b00-8eeb-11eb-9aba-b4f86261d125.png)

respondcreate commented:
Hey @luzfcb !

Just taking a look at this now and am having trouble running these tests with `tox` (everything fails). In your PR, can you please add some docs to the README that explain how to leverage tox in a local development environment (from installation to initiating the test matrix)?

Also, the travis CI test suite is still using to the current test matrix, can you also update `.travis.yml` to utilize this new tox setup to run tests and send coverage to coveralls?

Thanks so much for your contribution so far, looking forward to getting this ironed-out and merged in.

respondcreate commented:
Hey @luzfcb !

I was able to figure out the tox/travis/coveralls stuff. It's currently on the `github_actions_and_tox` branch of this repo. Still have some cleaning up to do but it's coming along!

luzfcb commented:
@respondcreate Sorry I saw this just now, I was offline all day.
I made some improvements to the tox configuration that I did initially, also converted the models and related files into a formal django app, and converted the test suite to be compatible with pytest

pickle api had never bitten me before, so it took me quite a while to discover and paying attention to __getstate __ / __ setstate__ pickle to fix compatibility.

Another thing I found is that pillow added a few more compile dependencies after 4.2

Anyway, this is what I have now:
![image](https://user-images.githubusercontent.com/807599/113463056-4265b180-93fa-11eb-8ac1-85d59e862f07.png)
